src/core/unit.c

```diff
@@ -4238,16 +4238,14 @@ int unit_patch_contexts(Unit *u) {
                                         return -ENOMEM;
                         }
 
-                if (MANAGER_IS_USER(u->manager) &&
-                    !ec->working_directory) {
-
+                if (ec->working_directory_home || (MANAGER_IS_USER(u->manager) && !ec->working_directory)) {
                         r = get_home_dir(&ec->working_directory);
```

Comment:
Hm? If `WorkingDirectory=~` is specified in a system unit, then it always depends on the home directory of root(UID0)??

> WorkingDirectory=
           Sets the working directory for executed processes. If set to "~", the home directory of the user
           specified in User= is used.

Reply:
Hmm, right. I'll drop the change.